homeassistant/components/gogogate2/common.py

```diff
@@ -0,0 +1,100 @@
+"""Common code for GogoGate2 component."""
+from datetime import timedelta
+import logging
+from typing import Awaitable, Callable, NamedTuple, Optional
+
+import async_timeout
+from gogogate2_api import GogoGate2Api
+from gogogate2_api.common import Door
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class StateData(NamedTuple):
+    """State data for a cover entity."""
+
+    config_unique_id: str
+    unique_id: Optional[str]
+    door: Optional[Door]
+
+
+class GogoGateDataUpdateCoordinator(DataUpdateCoordinator):
+    """Manages polling for state changes from the device."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        api: GogoGate2Api,
+        *,
+        name: str,
+        update_interval: timedelta,
+        update_method: Optional[Callable[[], Awaitable]] = None,
+        request_refresh_debouncer: Optional[Debouncer] = None,
+    ):
+        """Initialize the data update coordinator."""
+        DataUpdateCoordinator.__init__(
+            self,
+            hass,
+            logger,
+            name=name,
+            update_interval=update_interval,
+            update_method=update_method,
+            request_refresh_debouncer=request_refresh_debouncer,
+        )
+        self.api = api
+
+
+def get_data_update_coordinator(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> GogoGateDataUpdateCoordinator:
+    """Get an update coordinator."""
+    hass.data.setdefault(DOMAIN, {})
+    data_update_coordinators = hass.data[DOMAIN]
```

Comment:
The general practice is to store everything for the config entry under.  `hass.data[DOMAIN][config_entry.entry_id]`  Please see my earlier review comment.


```suggestion
    data_update_coordinator = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
```

Reply:
A facefalm reaction should really exist for when somebody catches my obvious oversight multiple times.